packages/flutter/lib/src/widgets/overscroll_indicator.dart

```diff
@@ -243,7 +243,7 @@ class _GlowingOverscrollIndicatorState extends State<GlowingOverscrollIndicator>
   final Map<bool, bool> _accepted = <bool, bool>{false: true, true: true};
 
   bool _handleScrollNotification(ScrollNotification notification) {
-    if (!widget.notificationPredicate(notification))
+    if (!widget.notificationPredicate(notification) || !notification.metrics.hasContentDimensions)
```

Comment:
This looks like a separate issue. It also does not have a test. I am wondering then what exactly creates this situation where there are null metrics? If we have null metrics but have triggered a scroll notification, that appears to be a symptom of a different issue.

Reply:
The issue explains how this situation happens. I think the real problem relates to the scroll metrics, as the properties could be null https://github.com/flutter/flutter/blob/59fc92168a4596b4f523ba5774a8c5cfece6cc31/packages/flutter/lib/src/widgets/scroll_metrics.dart#L135-L138
and this is not a safe getter
https://github.com/flutter/flutter/blob/59fc92168a4596b4f523ba5774a8c5cfece6cc31/packages/flutter/lib/src/widgets/scroll_metrics.dart#L146
And recently I found another issue relates to the `AppBar` in https://github.com/flutter/flutter/blob/59fc92168a4596b4f523ba5774a8c5cfece6cc31/packages/flutter/lib/src/material/app_bar.dart#L765
that triggers another issue.
It might be better to make scroll metrics safer by not using non-null semantic.

Reply:
Ah, I do not think this is an appropriate change to make then.

This would be a situation where we would want to embrace the yak shave and address the root of the problem, rather than trying to fix the areas where it becomes apparent. I am not sure if the scroll metrics properties being nullable is the issue, but it may be worth while to discuss the problem further in the linked issue, or write a design document proposing a change to scroll metrics so others could contribute feedback. 
There is a template you can use if you would like to write [a document like that here.](https://docs.google.com/document/d/1SFRO8U2toOlAaZ38dsuEU7Wm5fn41wvBCWKiwADqfmw/edit)

Reply:
Thanks for the reply. It's been a while :)

Yes, but changing the scroll metrics is like a breaking change. I will try to make another issue to get feedback. I think closing this PR is a better idea.

Reply:
SGTM! Thank you for the contribution!

Reply:
Thanks @Piinks, Not merged, but I will try another one ASAP :)